onnx/cpp2py_export.cc

```diff
@@ -625,6 +636,127 @@ PYBIND11_MODULE(onnx_cpp2py_export, onnx_cpp2py_export) {
   shape_inference.doc() = "Shape Inference submodule";
   py::register_exception<InferenceError>(shape_inference, "InferenceError");
 
+  py::class_<InferenceContext> inference_ctx(shape_inference, "InferenceContext", "Inference context");
+
+  inference_ctx.def("__get_attribute", [](InferenceContext* ctx, std::string name) {
+    if (ctx == nullptr) {
+      fail_shape_inference("Internal error: `ctx` is nullptr in `__get_attribute`");
+    }
+    auto attr = ctx->getAttribute(name);
+    if (attr == nullptr) {
+      fail_shape_inference("Internal error: `attr` is nullptr in `__get_attribute`");
+    }
+    std::string data;
+    attr->SerializeToString(&data);
+    return py::bytes(data);
+  });
+  inference_ctx.def("get_num_inputs", &InferenceContext::getNumInputs);
+  inference_ctx.def("has_input", &InferenceContext::hasInput);
+  inference_ctx.def("__get_input_type", [](InferenceContext* ctx, size_t index) {
+    if (ctx == nullptr) {
+      fail_shape_inference("Internal error: `ctx` is nullptr in `__get_input_type`");
+    }
+    auto type = ctx->getInputType(index);
+    if (type == nullptr) {
+      fail_shape_inference("Internal error: `type` is nullptr in `__get_input_type`");
+    }
+    std::string data;
+    type->SerializeToString(&data);
+    return py::bytes(data);
+  });
+  inference_ctx.def("__get_input_data", [](InferenceContext* ctx, size_t index) {
+    if (ctx == nullptr) {
+      fail_shape_inference("Internal error: `ctx` is nullptr in `__get_input_data`");
+    }
+    auto tensor = ctx->getInputData(index);
+    if (tensor == nullptr) {
+      fail_shape_inference("Internal error: `tensor` is nullptr in `__get_input_data`");
```

Comment:
Again, not an error. Signifies no static-constant-value is known for this input. Best to map this to `None` or some similar thing